src/refract/primitives.es6

```diff
@@ -0,0 +1,322 @@
+/*
+ * A refract element implementation with an extensible type registry.
```

Comment:
I wanted to circle back and add some comments along the way.

> Should the type registry really be global?

I wanted to put a couple things down for you to think about here. We had the concept of this registry where you registered your own types, but is there a way to do this so you don't have to create new type classes? In other words, if an instance of a Refract element knows it's a string, it will always behave like a string no matter name you give it. 

The idea of a registry gets even more complicated when you consider namespaces. Really, the element definitions propogate down through the tree, but not up. This means that if you include a namespace, the parent elements have no visibility of those definitions, though all the children will.

> if you need to push or set a simple type then call convertToType first. If you need to set a more complex element, then use your element class and construct it yourself.

Minim started out as a simple interface to JSON structures, and Refract was born from that. It makes for going from JSON to Refract easy.

```javascript
var obj = new Object
  .set('foo', 'bar')
  .set('a', 'b')
```

That makes sense to people new to these concepts. Having to include some convert function everywhere makes it somewhat less appealing to people coming from JSON, which uses simple dot notation.

I don't have any answers at this point, but I feel like getting adoption with this will require an interface that adds just a little complexity to what people are familiar with in the JSON world while providing the extra stuff Refract provides.

> Better error handling

This is one that I think has some really cool possibilities. My initial thought was to have an `Error` class that had all of the methods from all of the different kinds of elements (since JavaScript currently has no method missing). If you received an error, you created an instance and the rest of the chain worked, it just passed the error object all the way through. See this [PR here](https://github.com/smizell/minim/pull/21).

Kyle has brought up some interesting ideas around this, which would be to use Promises or [streams](https://github.com/jlongster/js-csp). This would be a change, but could provide an interface that can handle errors quite well.

---

Lots of rambling here. Hope it helps some! Excited to see the work and thought being put into this.